src/test/java/org/elasticsearch/test/discovery/ClusterDiscoveryConfiguration.java

```diff
@@ -126,22 +134,47 @@ public Settings node(int nodeOrdinal) {
                     .put("discovery.zen.ping.multicast.enabled", false);
 
             String[] unicastHosts = new String[unicastHostOrdinals.length];
-            String mode = nodeSettings.get("node.mode", InternalTestCluster.NODE_MODE);
-            if (mode.equals("local")) {
+            if (localMode) {
                 builder.put(LocalTransport.TRANSPORT_LOCAL_ADDRESS, "node_" + nodeOrdinal);
                 for (int i = 0; i < unicastHosts.length; i++) {
                     unicastHosts[i] = "node_" + unicastHostOrdinals[i];
                 }
             } else {
                 // we need to pin the node port & host so we'd know where to point things
-                builder.put("transport.tcp.port", basePort + nodeOrdinal);
+                builder.put("transport.tcp.port", unicastHostPorts[nodeOrdinal]);
                 builder.put("transport.host", "localhost");
-                for (int i = 0; i < unicastHosts.length; i++) {
-                    unicastHosts[i] = "localhost:" + (basePort + unicastHostOrdinals[i]);
+                for (int i = 0; i < unicastHostOrdinals.length; i++) {
+                    unicastHosts[i] = "localhost:" + (unicastHostPorts[unicastHostOrdinals[i]]);
                 }
             }
             builder.putArray("discovery.zen.ping.unicast.hosts", unicastHosts);
             return builder.put(super.node(nodeOrdinal)).build();
         }
+
+        protected static int[] unicastHostPorts(int basePort, int numHosts) {
+            int[] unicastHostPorts = new int[numHosts];
+
+            final int maxPort = basePort + 99;
```

Comment:
I think we can increase the 99 range to 1000 and also change calcBasePort to ignore the scope component as it is not relevant any more. I want if we should jut remember the last based port and continue from it (round robin-ing) to make in-JVM collisions less likely